azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -906,12 +907,18 @@ private void runExecutableNode(final ExecutableNode node) throws IOException {
 
     node.setStatus(Status.QUEUED);
 
+    // Attach Ramp Props if there is any desired properties
     String jobId = node.getId();
-    String jobType = node.getInputProps().getString("type");
-    Props rampProps = this.flow.getRampPropsForJob(jobId, jobType);
-    if (rampProps != null) {
-      this.logger.info("Selected Ramp Props in Flow : [" + rampProps + "]");
-      node.setRampProps(rampProps);
+    String jobType = Optional.ofNullable(node.getInputProps()).map(props -> props.getString("type")).orElse(null);
```

Comment:
if jobId could be null, then this line should be inside the if condition and jobtype's null check should be nested in it.

Reply:
No. JobId and jobType should not be Null. I just want to put more safe guard in case any accident which I don't know.

Reply:
Let me add warning when jobId or JobType is null, which should be impossible.